### PR TITLE
Fix xomachine/NESM#15

### PR DESCRIPTION
**Pull request by zintrepid**
Allow dynamic variants nested in periodics

### CODE REVIEW

**Comment by zintrepid** [comment]
The idea is to parenthesize the case statements rather than insert `result +=` code so that the statement remains the correct type and does not result in nested `result += result += x` code that does not compile.

File: tests/complex.nim [hunk 1]
```diff
@@ -87,3 +87,16 @@ suite "Complex tests":
...
+      type
+        VariantType = object
+          case a: uint8
+          of 1:
+            b: string
+          else:
+            c: uint8
+        PeriodicContainer = object
+          a: seq[VariantType]
+    check(true)
```
Comment (xomachine) [inline tests/complex.nim:102]: I think we need a serialization test here in addition to the compilation test

**Review by xomachine — COMMENTED** [review]
Thank you for your proposal. I will merge it via main repository on gitlab, so do not worry when the PR will be closed.

I will review other PRs too, but later. I do not have enough time right now

**Comment by zintrepid** [comment]
@xomachine  No problem. Would it be easier for me to use gitlab? I don't have an account there but perhaps I could  set one up. Also, I just want to say thank you for publishing this library! It made a project I was working on much simpler.

File: tests/complex.nim (hunk 1 above)
Comment (zintrepid) [reply tests/complex.nim:102]: I'll try to add one when I get time again. The code change affects all variants, so serialization works in at least the cases that are already tested.

**Comment by xomachine** [comment]
It is not a problem for me to move your commits to gitlab, so please use what you prefer. In my opinion it is worth to complete at least existing PRs on github

File: tests/complex.nim (hunk 1 above)
Comment (zintrepid) [reply tests/complex.nim:121]: I went ahead and added a basic test.

**Comment by xomachine** [comment]
Merged in 617a614ceab6bb863a6b57d8220d1c84d8d1fef6 e7be78656ed769403da0764aab27fface65163c6 c7fa25174bf926f8628b056091753cf97b8d1e2b

**Pull request closed without merging**